### PR TITLE
Feature/update donation person

### PR DESCRIPTION
**Pull request by stann1**
the backend part for: https://github.com/podkrepi-bg/frontend/issues/958

### CODE REVIEW

**Review by igoychev — CHANGES_REQUESTED** [review]
great overall, pls check my comments below

File: apps/api/src/donations/donations.service.ts [hunk 1]
```diff
@@ -301,36 +301,65 @@ export class DonationsService {
...
+      }
 
-      if (oldDonationStatus === DonationStatus.succeeded) {
-        throw new Error('Succeded donations cannot be updated.')
+      if (
+        currentDonation.status === DonationStatus.succeeded &&
+        updatePaymentDto.status &&
+        updatePaymentDto.status !== DonationStatus.succeeded
+      ) {
+        throw new BadRequestException('Succeeded donations cannot be updated.')
```
[the remark this answers is not in the document]
Comment (stann1) [reply apps/api/src/donations/donations.service.ts:324]: this was the original message, and I actually like it - short and to the point

File: apps/api/src/donations/donations.service.ts [hunk 2]
```diff
@@ -301,36 +301,65 @@ export class DonationsService {
...
+        updatePaymentDto.targetPersonId &&
+        currentDonation.personId !== updatePaymentDto.targetPersonId
+      ) {
+        const targetDonor = await this.prisma.person.findFirst({
+          where: { id: updatePaymentDto.targetPersonId },
+        })
+        if (!targetDonor) {
+          throw new NotFoundException(
+            `Update failed. No person found with ID: ${updatePaymentDto.targetPersonId}`,
+          )
```
[the remark this answers is not in the document]
Comment (stann1) [reply apps/api/src/donations/donations.service.ts:339]: true, but then the user will get a system unhandled exception, this way it is much more user friendly

File: apps/api/src/donations/donations.service.ts [hunk 3]
```diff
@@ -301,36 +301,65 @@ export class DonationsService {
...
         },
       })
 
       if (updatePaymentDto.status === DonationStatus.succeeded) {
-        await this.vaultService.incrementVaultAmount(donation.targetVaultId, donation.amount)
+        await this.vaultService.incrementVaultAmount(
+          currentDonation.targetVaultId,
+          currentDonation.amount,
+        )
       }
```
[the remark this answers is not in the document]
Comment (stann1) [reply apps/api/src/donations/donations.service.ts:357]: I agree, will change it

File: apps/api/src/donations/donations.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (igoychev) [reply apps/api/src/donations/donations.service.ts:339]: What do you think about handling the exception. Just found that Prisma has an error code for this case P2003 and the exception docs show how to handle it: https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors

**Review by igoychev — APPROVED** [review]

**Pull request merged**